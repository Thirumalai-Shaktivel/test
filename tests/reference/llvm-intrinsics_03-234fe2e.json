{
    "basename": "llvm-intrinsics_03-234fe2e",
    "cmd": "lfortran --show-llvm {infile} -o {outfile}",
    "infile": "tests/../integration_tests/intrinsics_03.f90",
    "infile_hash": "eda277452cad2d7bc3ef98963677fc276c09ce4d517756982a0d921d",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-intrinsics_03-234fe2e.stdout",
    "stdout_hash": "c9f5e3a1f8af2a53b33d8b8540b1327237c83ddc4f481fc077c44632",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}