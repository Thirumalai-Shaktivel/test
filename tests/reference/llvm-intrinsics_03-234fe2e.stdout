; ModuleID = 'LFortran'
source_filename = "LFortran"

@0 = private unnamed_addr constant [4 x i8] c"%f\0A\00", align 1

define float @__module_lfortran_intrinsic_math_scos(float* %x) {
.entry:
  %r = alloca float, align 4
  %0 = load float, float* %x, align 4
  %1 = call float @_lfortran_scos(float %0)
  store float %1, float* %r, align 4
  %2 = load float, float* %r, align 4
  ret float %2
}

declare float @_lfortran_scos(float)

define i32 @main() {
.entry:
  %x = alloca float, align 4
  %0 = alloca float, align 4
  store float 9.500000e+00, float* %0, align 4
  %1 = call float @__module_lfortran_intrinsic_math_scos(float* %0)
  store float %1, float* %x, align 4
  %2 = load float, float* %x, align 4
  %3 = fpext float %2 to double
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @0, i32 0, i32 0), double %3)
  ret i32 0
}

declare void @_lfortran_sin(float, float*)

declare void @_lfortran_cos(float, float*)

declare void @_lfortran_tan(float, float*)

declare void @_lfortran_sinh(float, float*)

declare void @_lfortran_cosh(float, float*)

declare void @_lfortran_tanh(float, float*)

declare void @_lfortran_asin(float, float*)

declare void @_lfortran_acos(float, float*)

declare void @_lfortran_atan(float, float*)

declare void @_lfortran_asinh(float, float*)

declare void @_lfortran_acosh(float, float*)

declare void @_lfortran_atanh(float, float*)

declare void @_lfortran_printf(i8*, ...)

